oc_ldap_user_locker/locker.py

```diff
@@ -196,24 +234,106 @@ def _process_single_user(self, ldap_c, user_dn):
         # now check the time attributes specified in the conf and find out the nearest one
         # note that 'tzinfo' is to be discarged because of possible datetime exception while
         #   subtracting them
-        _lock = self._check_lock_conf(_user_rec, _conf['days_valid'], _conf['time_attributes'])
+        _lock_date = self._get_account_lock_date(_user_rec, _conf['days_valid'], _conf['time_attributes'])
 
-        if not _lock:
+        if not _lock_date:
+            # should never happen
+            logging.debug("Account '%s' is not to be locked ever", _user_rec.get_attribute('cn'))
             return
 
+        logging.debug("Account lock date for '%s': '%s'" % (
+            _user_rec.get_attribute('cn'), _lock_date.isoformat(sep=" ")))
+
+        _days_before_lock = self._get_days_before_lock(_lock_date)
+        logging.debug("Days before lock account '%s': %d" % (
+            _user_rec.get_attribute('cn'), _days_before_lock))
+
+        # check lock e-mail notifications
+        self._check_lock_notifications(
+                _user_rec, _conf, lock_date=_lock_date, days_before_lock=_days_before_lock)
+
+        if _days_before_lock > 0:
+            logging.debug("Is not the time to lock '%s', returning" % _user_rec.get_attribute('cn'))
```

Comment:
I may be picking, but I'd rather write 'returning None', or add a period. Looks like debug is missing, what specifically it is returning.